Feature/preprocessing for embedding component

Adem-Khachnaoui opened the pull request:
- textblocks are now pre-processed before being embedded.
- No code duplication between embedding and segmentation

jpbernius commented on docker-compose.yml:
```diff
@@ -11,6 +11,8 @@ services:
     ports:
       - 8000:8000
     working_dir: /usr/src/app
+    volumes:
+      - ./text_preprocessing:/usr/src/app/src/text_preprocessing
```
I do not think this is enough. The Docker Containers will no longer work without the source code being available.  Can we copy it into the Docker Image? Maybe using a symlink?

jpbernius commented on docker-compose.yml:
```diff
@@ -24,6 +26,8 @@ services:
     ports:
       - 8001:8000
     working_dir: /usr/src/app
+    volumes:
+      - ./text_preprocessing:/usr/src/app/src/text_preprocessing
```
I do not think this is enough. The Docker Containers will no longer work without the source code being available.  Can we copy it into the Docker Image? Maybe using a symlink?

Adem-Khachnaoui replied:
Sorry I don't get it.
I thought the volume is copied at start-up time so no need for the source to be available on runtime

jpbernius replied:
Volumes are mounted live into the container (which es great e.g. for development), but not copied.

linusmichel replied:
@Adem-Khachnaoui that would mean the container is not able to run "standalone" without the volume, hence without the repository. But on our processing-nodes, we won't have the repo cloned on there to run the containers. So we need to somehow make sure the ./text_preprocessing directory is copied to the container at buildtime, not only at runtime.

Adem-Khachnaoui replied:
Problem solved! please have a look

Adem-Khachnaoui commented:
This PR is again ready for review. Please have a look !

linusmichel reviewed (dismissed):
I ran into an issue with the new build context, if mongodb data exists in the root directory. Including
`data`
in the `.dockerignore` file resolves this. I added this change to the branch.

Rest looks good to me and works!

Maybe a few more comments in the code of the "benchmark"-stuff would help :)

Adem-Khachnaoui commented:
> I ran into an issue with the new build context, if mongodb data exists in the root directory. Including
> `data`
> in the `.dockerignore` file resolves this. I added this change to the branch.
> 
> Rest looks good to me and works!
> 
> Maybe a few more comments in the code of the "benchmark"-stuff would help :)

 @linusmichel :
Thanks for the Review, 
I added some comments and a readme in the benchmark (as part of https://github.com/ls1intum/Athene/pull/24)
